app/models/concerns/message_export_operations.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module MessageExportOperations
+  extend ActiveSupport::Concern
+
+  included do
+    def prepare_message_export
+      ::Zip::OutputStream.write_buffer do |zip|
+        prepare_original_objects(zip)
+        prepare_print_objects(zip)
+      end.string
+    end
+
+    private
+
+    def prepare_original_objects(zip)
+      objects.each do |message_object|
+        zip.put_next_entry("na_pravne_ucely/#{MessageObjectHelper.displayable_name(message_object)}")
+        zip.write(message_object.content)
+      end
+    end
+
+    def prepare_print_objects(zip)
+      objects.each do |message_object|
+        pdf_content = message_object.prepare_pdf_visualization
+
+        next unless pdf_content
+
+        zip.put_next_entry("na_tlac/#{MessageObjectHelper.pdf_name(message_object)}")
```

Comment:
tie vizualizovane by som dal priamo do rootu a tie "originaly" by som nazval `originaly` - lebo v govbox domene toto nedava zmysel. "na pravne ucely" to totiz nemusi byt vzdy.

Reply:
takze v:
- `originaly` podpriecinku bude vsetko,
- roote iba PDF vizualizacie?

Reply:
pozeram na to takto - mame v core domene spravu, ktora ma form a potom nejake prilohy, ktore mozu mat vo vnutri nieco.

Na najvyssej urovni by som chcel veci, ktore nejako viem vidiet, cize asi ten form (pdf) a vnutra vselijakych asice (ak tam je form tak pdf). vo foldri `originaly` by bolo vsetko bez transformacie. cize xml/form a asice (top level dokumenty)